lib/auth/scram.js

```diff
@@ -143,7 +161,7 @@ class ScramSHA extends AuthProvider {
 
       processedPassword = saslprep ? saslprep(password) : password;
     } else {
-      processedPassword = passwordDigest(username, password);
+      processedPassword = passwordDigest(username, password, callback);
```

Comment:
rather than changing the `passwordDigest` function, can we instead wrap this in a `try`/`catch`?

```js
} else {
  try {
    processedPassword = passwordDigest(username, password);
  } catch (e) {
    return callback(e);
  }
}
```

Reply:
I prefer this approach as well, it requires less passing of `callback` around. It's a little bit weird that the proposed approach above would use `callback` to return errors, but otherwise return a value.